cartridge/roles/coordinator.lua

```diff
@@ -195,47 +194,53 @@ local function take_control(uri)
         end
 
         if pcall(fiber.status, control_fiber) == 'dead'
-        or not errors.netbox_call(conn, 'acquire_lock',
-            lock_args, {timeout = vars.options.NETBOX_CALL_TIMEOUT}
-        ) then
+        or not session:acquire_lock(lock_args)
+        then
             break
         end
     until not pcall(fiber.testcancel)
 
-    pcall(conn.close, conn)
+    session:drop()
+    log.info('Lock released')
     pcall(fiber.cancel, control_fiber)
-    vars.conn = nil
 
-    log.info('Lock released')
     return true
 end
 
-local function connect_loop(uri)
-    checks('string')
+local function take_control_loop(client)
+    checks('stateboard_client')
+
     repeat
         local t1 = fiber.time()
-        local ok, err = CoordinatorError:pcall(take_control, uri)
+        local ok, err = CoordinatorError:pcall(take_control, client)
         local t2 = fiber.time()
 
         if ok == nil then
+            fiber.testcancel()
             log.error('%s', type(err) == 'table' and err.err or err)
         end
 
         if ok ~= true then
             fiber.sleep(t1 + vars.options.RECONNECT_PERIOD - t2)
         end
-
     until not pcall(fiber.testcancel)
 end
 
 local function stop()
-    if vars.connect_fiber == nil then
-        return
-    elseif vars.connect_fiber:status() ~= 'dead' then
-        vars.connect_fiber:cancel()
+    if vars.connect_fiber ~= nil then
+        pcall(fiber.cancel, vars.connect_fiber)
+        vars.connect_fiber = nil
+    end
+
+    if vars.session ~= nil then
```

Comment:
I suppose that `vars.client:drop_session()` should be enough to drop session. I don't understand such duplication.